packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -454,77 +248,94 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		storageAddOnsForPlan,
 	} );
 
-	const onCtaClick = useActionCallback( {
+	const priceString = formatCurrency(
+		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
+		currencyCode || 'USD',
+		{
+			stripZeros: true,
+			isSmallestUnit: true,
+		}
+	);
+
+	const isTrialPlan =
+		currentSitePlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY ||
+		currentSitePlanSlug === PLAN_MIGRATION_TRIAL_MONTHLY ||
+		currentSitePlanSlug === PLAN_HOSTING_TRIAL_MONTHLY;
+
+	// If the current plan is on a higher-term but lower-tier, then show a "Contact support" button.
+	// TODO: Consider the structure of this condition. Should we move it out of this component?
+	const higherTermButLowerTierPlan =
+		availableForPurchase &&
+		currentSitePlanSlug &&
+		! current &&
+		! isTrialPlan &&
+		currentPlanBillingPeriod &&
+		billingPeriod &&
+		currentPlanBillingPeriod > billingPeriod;
```

Comment:
This is the same expression with line 266 in `use-generate-action-hook`. It shouldn't live here separately. It was meant for showing the "contact support" button (now passed from the other file). It's just also used to make the button disabled. So I guess it's safe to remove and simply pass a "disabled" `status` from useAction hook.

Reply:
Should be addressed in https://github.com/Automattic/wp-calypso/pull/89926/commits/ac21385b6a31bcc09044b471f1b981b85b66479d (let's double check pls)

Reply:
Ty! Double checking now

Reply:
So it seems like this condition may never be met with our current UI. The new plan term selector never allows selection of lower terms. Ex. if a two year plan is purchased, the one year and monthly plan terms are no longer available options in the term selector dropdown.

This is something I'd want to have a chance to take a closer look at before ripping out though, so I'm leaving it for the follow-up.

Reply:
Good call, although term-selector won't block user from accessing `/plans/[ 2y-site ]/yearly` directly.